airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/connector_runner.py

```diff
@@ -44,37 +44,38 @@ def output_folder(self) -> Path:
     def input_folder(self) -> Path:
         return self._volume_base / f"run_{self._runs}" / "input"
 
-    def _prepare_volumes(self, config: Optional[Mapping], state: Optional[Mapping], catalog: Optional[ConfiguredAirbyteCatalog]):
+    def _prepare_image(self, config: Optional[Mapping], state: Optional[Mapping], catalog: Optional[ConfiguredAirbyteCatalog]):
         self.input_folder.mkdir(parents=True)
-        self.output_folder.mkdir(parents=True)
+
+        current_image_tag = self._image.tags[0]
+        new_image_tag = current_image_tag + "-with-secrets"
+        dockerfile = f"FROM {current_image_tag}\n RUN mkdir /data\n"
```

Comment:
I think we could get away from this if we got rid of dockerfiles and instead used dagger api to do this behavior, no? Though that is probably a much larger refactor than this approach

Reply:
We use "dagger in dagger" for our own CI, so we can definitely explore this in the future 😃

Reply:
The connector-acceptance-test project does not need to be patched now that I can use the `insecure_root_capabilities` flag.